### PR TITLE
fix: result types

### PR DESCRIPTION
**Pull request by ycjcl868**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by ycjcl868** [comment]
类型有点问题，先不合

File: components/result/index.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ export type ResultStatusType = ExceptionStatusType | keyof typeof IconMap;
 
 export interface ResultProps {
   icon?: React.ReactNode;
-  status: ResultStatusType;
+  status?: ResultStatusType;
```
Comment (ztplz) [inline components/result/index.tsx]: `status?` === `ResultStatusType || undefined`  看了一下  后面status直接解构的  用到的地方要加non-null assertion operator

**Review by ztplz — APPROVED** [review]

File: components/result/index.tsx (hunk 1 above)
Comment (ycjcl868) [reply components/result/index.tsx]: status 可以不传，我加一把 asset

File: components/result/index.tsx [hunk 2]
```diff
@@ -46,7 +46,7 @@ const ExceptionStatus = Object.keys(ExceptionMap);
 const renderIcon = (prefixCls: string, { status, icon }: ResultProps) => {
   const className = classnames(`${prefixCls}-icon`);
 
-  if (ExceptionStatus.includes(status)) {
+  if (ExceptionStatus.includes(status as ResultStatusType)) {
```
Comment (ztplz) [inline components/result/index.tsx]: 不需要assertion 直接非空就行了

**Pull request merged**